src/openapi3/handleJson.ts

```diff
@@ -11,68 +12,83 @@ export const removeTrailingSlash = (url: string): string => {
   return url;
 };
 
-export default function (
+/* Assumptions:
+  RESTful APIs typically have two paths per resources -  a `/noun` path and a 
+  `/noun/{id}` path. `getResources` strips out the latter, allowing us to focus
+  on the former.
+
+  In OpenAPI3, the `/noun` path will typically have a `get` action, that 
+  probably accepts parameters and would respond with a array of objects, 
+  described in the `items` field.
+*/
+
+export default async function (
   response: OpenAPIV3.Document,
   entrypointUrl: string
-): Resource[] {
-  const paths = getResources(response.paths);
+): Promise<Resource[]> {
+  const results: ResolvedRefsResults = await jsonRefs.resolveRefs(response);
+  const document: OpenAPIV3.Document = results.resolved as OpenAPIV3.Document;
+
+  const paths = getResources(document.paths);
+
+  let serverUrlOrRelative = "/";
+  if (document.servers) {
+    serverUrlOrRelative = document.servers[0].url;
+  }
+
+  const serverUrl = new URL(serverUrlOrRelative, entrypointUrl).href;
 
-  const resources = paths.map((item) => {
+  const resources: Resource[] = [];
+
+  paths.forEach((item) => {
     const name = item.replace(`/`, ``);
-    const url = removeTrailingSlash(entrypointUrl) + item;
-    const firstMethod = Object.keys(
-      response.paths[item]
-    )[0] as keyof OpenAPIV3.PathItemObject;
-    const responsePathItem = response.paths[item][
-      firstMethod
-    ] as OpenAPIV3.OperationObject;
-
-    if (!responsePathItem.tags) {
-      throw new Error(); // @TODO
-    }
+    const url = removeTrailingSlash(serverUrl) + item;
 
-    const title = responsePathItem.tags[0];
+    const method = document.paths[item].get as OpenAPIV3.OperationObject;
```

Comment:
Why not keeping the "first method" approach like before. A resource *could* have only a post operation, doesn't it?

Reply:
I was a bit confused about the `firstMethod` stuff, to be honest - it seemed to just grab the first verb it could find, no matter what it was. Again, Open API 3 is an awkward match for a resource-based approach, but it seemed to me that the minimum for an admin interface for would to at least list the resources, which would be a `get` against the 'resource' path.

Reply:
OK I understand. What do you think of having a fallback on the other operations if the `get` is not defined?

Reply:
I'm open to it, but I'm not sure how it would work. How would we identify the right list verb if it's not a `get`? I don't think that, say, "it returns a list of items" is a clear indicator that a `put` is the right mechanism for getting the items for a resource. If it's not RESTful, I think it might be too difficult to guess with any degree of success... 

How were you imagining it would work?

Reply:
I understand your point of view, but here's mine:
- `api-doc-parser` is not only to be used by the admin, so it can make sense to have the resources even if there is no `get` operation
- even in the context of the admin, sometimes you may want to create a resource but not necessarily list it.
WDYT?

Reply:
It'd be great to be able to include those other resources - could you sketch out how you imagine it'd work? (I'm guessing that's what the `firstMethod` approach is about - grabbing the first verb available for every 'resource' path and building a `resource` from that.)

The problem I'd foresee with that are:
1. We could reasonably end up with a lot of junk `resources`, built from paths that don't represent resources (Open API 3 has no requirements or recommendations about the resource-ness/RESTfulness of the paths involved.)
2. We'd then have to filter those resources out at a higher level - we often wouldn't be able to automatically include all generated resources in, say the `admin` platform (annoying, but not a deal-breaker.)
3. In Open API 3 the schema information we want is defined per verb, so I'm concerned it might be quite unreliable, using the schema from one verb for another. (This is generally a problem trying to map Open API 3 to this 'resource' pattern - me assuming the `get` schema is the right one for 'create', etc, was also a hack, of course.)

Regarding creating a resource without listing it, is that something the `admin` platform handles? How does it know that a resource is 'create-able', but not 'listable'? (I guess part of my confusion is that I can't find any documentation for `api-doc-parser`, so I'm feeling around in the dark a bit, trying to map Open API 3 correctly to it's intermediate format.)

Reply:
Sure OpenAPI is about paths, not resources. But what guarantees you that if there is a `get` operation, the corresponding path corresponds to a resource?
I think we should just assume that every path corresponds to a resource and if the final user has added a custom path, it is its responsibility to filter it (like you said it a higher level).

For the schema information, IMO the aim is to have as much information as possible for the parsed resource. So we would probably need to use all the verbs and to choose the most likely information.

For now it's not correctly handled in the `admin` part, but we could set the `list` prop to `false` (here: https://github.com/api-platform/admin/blob/555ed56f4bb3eea17b30c0150d38c6dde274f576/src/ResourceGuesser.js#L14) if there is no `Operation` with a `GET` method.